cdap-proto/src/main/java/co/cask/cdap/proto/security/Authorizable.java

```diff
@@ -0,0 +1,199 @@
+   * or applicationId which has version is passed to {@link Authorizable#fromEntityId(EntityId)} then the version
+   * will be silently dropped to construct the authorizable.
+   * </li>
+   * </ol>
+   *
+   * @param entityString the {@link EntityId#toString()} of the entity which may or may not contains wildcards(? or *)
+   */
+  public Authorizable(String entityString) {
+    if (entityString == null || entityString.isEmpty()) {
+      throw new IllegalArgumentException("A valid authorizable must be provided.");
+    }
+
+    String[] typeAndId = entityString.split(EntityId.IDSTRING_TYPE_SEPARATOR, 2);
+    if (typeAndId.length != 2) {
+      throw new IllegalArgumentException(
+        String.format("Expected type separator '%s' to be in the ID string: %s", EntityId.IDSTRING_TYPE_SEPARATOR,
+                      entityString));
+    }
+    String typeString = typeAndId[0];
+    EntityType type = EntityType.valueOf(typeString.toUpperCase());
+    String idString = typeAndId[1];
+    List<String> idParts = Arrays.asList(EntityId.IDSTRING_PART_SEPARATOR_PATTERN.split(idString));
+    checkParts(type, idParts, idParts.size() - 1);
+    entityType = type;
+  }
+
+  /**
+   * Created an {@link Authorizable} which represents the given entityId.
+   * Note: CDAP Authorization does not support authorization on versions of {@link co.cask.cdap.proto.id.ApplicationId}
+   * and {@link co.cask.cdap.proto.id.ArtifactId}. If an artifactId or applicationId which has version is passed to
+   * then the version will be silently dropped to construct the authorizable.
+   *
+   * @param entityId the entity
+   * @return {@link Authorizable} representing the entity
+   */
+  public static Authorizable fromEntityId(EntityId entityId) {
+    if (entityId == null) {
+      throw new IllegalArgumentException("EntityId is required.");
+    }
+    String entity = entityId.toString();
+    // drop the version for artifact or application
+    if (entityId.getEntityType().equals(EntityType.ARTIFACT) ||
+      entityId.getEntityType().equals(EntityType.APPLICATION) || entityId.getEntityType().equals(EntityType.PROGRAM)) {
+      int versionStartIndex = entity.indexOf(EntityId.IDSTRING_PART_SEPARATOR,
+                                             entity.indexOf(EntityId.IDSTRING_PART_SEPARATOR) + 1);
+      int versionEndIndex = entity.length();
+      if (entityId.getEntityType().equals(EntityType.PROGRAM)) {
+        // for programs versions doesn't end at the end of the entity string as there is program type and name
+        // afterwards
+        versionEndIndex = entity.lastIndexOf(EntityId.IDSTRING_PART_SEPARATOR,
+                                             entity.lastIndexOf(EntityId.IDSTRING_PART_SEPARATOR) - 1);
+      }
+      // remove the version from the entity string
+      String version = entity.substring(versionStartIndex, versionEndIndex);
+      entity = entity.replace(version, "");
```

Comment:
I am not able to understand this comment. If by index it means position then that's what being done now. 
If by index it means index or version after split that looks brittle since versions can have 1+ . in it. So currently we take elements from beginning and end which we want and leave the residue as version.